modules/apps/transfer/ibc_module.go

```diff
@@ -119,15 +119,15 @@ func (im IBCModule) OnChanOpenTry(
 		return "", err
 	}
 
-	if !slices.Contains(types.SupportedVersions, counterpartyVersion) {
-		return types.V2, nil
-	}
-
 	// OpenTry must claim the channelCapability that IBC passes into the callback
 	if err := im.keeper.ClaimCapability(ctx, chanCap, host.ChannelCapabilityPath(portID, channelID)); err != nil {
 		return "", err
 	}
 
+	if !slices.Contains(types.SupportedVersions, counterpartyVersion) {
+		return types.V2, nil
+	}
+
```

Comment:
This was the only change that happened, there was actually a bug where the capability wasn't being claimed correctly which happened when the default V2 was returned. We now claim capability before this return.